providers/test/Test.fst

```diff
@@ -359,126 +374,347 @@ let test_dh () : St unit =
-  (if avx2 then C.String.print !$"AVX2 ")
+let ts_snoc (ts: test_set) (c: config)  : Tot test_set =
+  ts `ts_append` ts_one c
 
-#push-options "--z3rlimit 16"
+inline_for_extraction
+noextract
+let ts_cons (c: config) (ts: test_set) : Tot test_set =
+  ts_one c `ts_append` ts
 
 inline_for_extraction
 noextract
-let test_all_on_config (p: platform) (avx avx2: bool) : St unit =
-  C.String.print !$"=================================================\n";
-  print_config p avx avx2;
-  C.String.print !$"\n";
-  AC.init ();
-  let has_avx = AC.has_avx () in
-  let has_avx2 = AC.has_avx2 () in
-  if avx && not has_avx
-  then
-    C.String.print !$"skipping, AVX wanted but not present\n"
-  else if avx2 && not has_avx2
-  then
-    C.String.print !$"skipping, AVX2 wanted but not present\n"
-  else if not (static_config_ok p)
-  then
-    C.String.print !$"skipping, not in static config\n"
-  else begin
-    (if p <> Vale then AC.disable_vale ());
-    (if p <> HACL then AC.disable_hacl ());
-    (if p <> OpenSSL then AC.disable_openssl ());
-    (if p <> BCrypt then AC.disable_bcrypt ());
-    (if not avx then AC.disable_avx ());
-    (if not avx2 then AC.disable_avx2 ());
-    let wants_vale = AC.wants_vale () in
-    let wants_hacl = AC.wants_hacl () in
-    let wants_openssl = AC.wants_openssl () in
-    let wants_bcrypt = AC.wants_bcrypt () in 
-    let wants_avx = AC.has_avx () in
-    let wants_avx2 = AC.has_avx2 () in
-    if not (
-      (wants_vale = (p = Vale)) &&
-      (wants_hacl = (p = HACL)) &&
-      (wants_openssl = (p = OpenSSL)) &&
-      (wants_bcrypt = (p = BCrypt)) &&
-      (wants_avx = avx) &&
-      (wants_avx2 = avx2)
-    )
-    then
-      C.Failure.failwith !$"autoconfig is inconsistent"
-    else
-      test_all_body (fun s -> print_config p avx avx2; C.String.print s)
-  end
+let external_test_set =
+  ts_nil
+  `ts_snoc` (OpenSSL, f_none)
+  `ts_snoc` (BCrypt, f_none)
 
-#pop-options
+inline_for_extraction
+noextract
+let poly1305_test_set =
+  (Hacl, f_avx2) `ts_cons` (
+  (Hacl, f_avx) `ts_cons` (
+  (Hacl, f_none) `ts_cons` (
+  (Vale, f_none) `ts_cons` (
+  external_test_set))))
 
 inline_for_extraction
 noextract
-let test_all_on_platform_avx (p: platform) (avx: bool) : St unit =
-  test_all_on_config p avx false;
-  test_all_on_config p avx true
+let curve25519_test_set =
+  (Hacl, f_bmi2 `f_concat` f_adx) `ts_cons` (
+  (Hacl, f_none) `ts_cons`
+  external_test_set)
 
 inline_for_extraction
 noextract
-let test_all_on_platform (p: platform) : St unit =
-  test_all_on_platform_avx p false;
-  test_all_on_platform_avx p true
+let aead_gcm_test_set =
+  (Vale, f_aesni `f_concat` f_avx) `ts_cons` (
+  ts_nil)
+
+inline_for_extraction
+noextract
+let chacha20poly1305_test_set =
+  (Hacl, f_none) `ts_cons`
+  external_test_set
+
+inline_for_extraction
+noextract
+let hash_test_set =
+  (Vale, f_none) `ts_cons` (
+  (Vale, f_shaext) `ts_cons` (
+  (Hacl, f_none) `ts_cons` (
+  external_test_set)))
```

Comment:
we no longer use openssl or bcrypt for hashes; is `external_test_set` still needed here?